File: code/backend/python/app/core/ports/dev/test_data_cleanup_port.py
from abc import ABC, abstractmethod

from app.core.models.dev.cleanup_result import CleanupResult


class TestDataCleanupPort(ABC):
    """Inbound port for the dev-only test-data cleanup use case."""

    # Tells pytest this is not a test class despite the "Test" name prefix.
    __test__ = False

    @abstractmethod
    def cleanup_test_data(self) -> CleanupResult:
        """Delete every guest and match created by automated test runs."""
        ...

File: code/backend/php/src/Core/Port/Dev/TestDataCleanupPort.php
<?php

namespace Games\Paths\Core\Port\Dev;

use Games\Paths\Core\Domain\Dev\CleanupResult;

/**
 * Inbound port for the dev-only test-data cleanup use case.
 */
interface TestDataCleanupPort
{
    /**
     * Delete every guest and match created by automated test runs.
     */
    public function cleanupTestData(): CleanupResult;
}
